src/features/game/quiz/GameQuizManager.ts

```diff
@@ -0,0 +1,214 @@
+    const choices = question.options;
+    const quizContainer = new Phaser.GameObjects.Container(scene, 0, 0);
+
+    const quizPartitions = Math.ceil(choices.length / 5);
+    const quizHeight = choices.length;
+
+    //Create quiz box contains quiz questions
+    const quizQuestionBox = createDialogueBox(scene);
+
+    //Create text writer to display quiz questions
+    const quizQuestionWriter = createTypewriter(scene, questionTextStyle);
+    const lineWithName = question.question.replace('{name}', this.getUsername());
+    quizQuestionWriter.changeLine(lineWithName);
+
+    GameGlobalAPI.getInstance().storeDialogueLine(lineWithName, question.speaker);
+
+    //Generate UI components for quizzes
+    const header = new Phaser.GameObjects.Text(
+      scene,
+      screenSize.x - QuizConstants.textPad,
+      QuizConstants.y,
+      questionPrompt,
+      quizTextStyle
+    ).setOrigin(1.0, 0.0);
+
+    const quizHeaderBg = new Phaser.GameObjects.Rectangle(
+      scene,
+      screenSize.x,
+      QuizConstants.y - QuizConstants.textPad,
+      QuizConstants.width * quizPartitions,
+      header.getBounds().bottom * 0.5 + QuizConstants.textPad,
+      HexColor.darkBlue,
+      0.8
+    ).setOrigin(1.0, 0.0);
+
+    const quizBg = new Phaser.GameObjects.Rectangle(
+      scene,
+      screenSize.x,
+      QuizConstants.y - QuizConstants.textPad,
+      QuizConstants.width * quizPartitions,
+      quizHeaderBg.getBounds().bottom * 0.5 + (quizHeight + 0.5) * QuizConstants.yInterval,
+      HexColor.lightBlue,
+      0.2
+    ).setOrigin(1.0, 0.0);
+
+    quizContainer.add([
+      quizBg,
+      quizHeaderBg,
+      header,
+      quizQuestionBox,
+      quizQuestionWriter.container
+    ]);
+
+    const buttonPositions = calcListFormatPos({
+      numOfItems: choices.length,
+      xSpacing: 0,
+      ySpacing: QuizConstants.yInterval
+    });
+
+    GameGlobalAPI.getInstance().addToLayer(Layer.Dialogue, quizContainer);
+
+    //Create options for users to select
+    const activateQuizContainer: Promise<any> = new Promise(resolve => {
+      quizContainer.add(
+        choices.map((response, index) =>
+          createButton(scene, {
+            assetKey: ImageAssets.mediumButton.key,
+            message: response.text,
+            textConfig: QuizConstants.textConfig,
+            bitMapTextStyle: quizOptStyle,
+            onUp: async () => {
+              quizContainer.destroy();
+              const isCorrect = index === question.answer ? 1 : 0;
+              if (response.reaction) {
+                await this.showReaction(response.reaction);
+              }
+              resolve(isCorrect);
+            }
+          }).setPosition(
+            screenSize.x -
+              QuizConstants.width / 2 -
+              QuizConstants.width * (quizPartitions - Math.floor(index / 5) - 1),
+            (buttonPositions[index][1] % (5 * QuizConstants.yInterval)) +
+              quizHeaderBg.getBounds().bottom +
+              75
```

Comment:
Where did the `75` come from? Should we make this (and any others) a constant?

Reply:
Comment addressed: commit e33cb15ab4d7c8c1ff47a4a8b1782b51747d523b